usr/share/rear/output/default/95_copy_result_files.sh

```diff
@@ -17,40 +17,31 @@ case "$scheme" in
     (nfs|cifs|usb|file|sshfs|ftpfs|davfs)
         # if called as mkbackuponly then we just don't have any result files.
         if test "$RESULT_FILES" ; then
-            Log "Copying files '${RESULT_FILES[@]}' to $scheme location"
-            cp $v "${RESULT_FILES[@]}" "${opath}/" >&2
-            StopIfError "Could not copy files to $scheme location"
+            Log "Copying result files '${RESULT_FILES[@]}' to $opath at $scheme location"
+            cp $v "${RESULT_FILES[@]}" "${opath}/" >&2 || Error "Could not copy result files to $opath at $scheme location"
         fi
-        echo "$VERSION_INFO" >"${opath}/VERSION"
-        StopIfError "Could not create VERSION file on $scheme location"
-
-        cp $v $(get_template "RESULT_usage_$OUTPUT.txt") "${opath}/README" >&2
-        StopIfError "Could not copy usage file to $scheme location"
-
+        echo "$VERSION_INFO" >"${opath}/VERSION" || Error "Could not create ${opath}/VERSION file at $scheme location"
+        cp $v $( get_template "RESULT_usage_$OUTPUT.txt" ) "${opath}/README" >&2 || Error "Could not copy usage file to ${opath}/README at $scheme location"
         # REAR_LOGFILE=/var/log/rear/rear-$HOSTNAME.log (name set by main script)
-        cat "$REAR_LOGFILE" >"${opath}/rear.log"
-        StopIfError "Could not copy $REAR_LOGFILE to $scheme location"
+        cat "$REAR_LOGFILE" >"${opath}/rear.log" || Error "Could not copy $REAR_LOGFILE to ${opath}/rear.log at $scheme location"
     ;;
 
     (fish|ftp|ftps|hftp|http|https|sftp)
-    LogPrint "Copying files '${RESULT_FILES[*]}' to $scheme location"
+    LogPrint "Copying result files '${RESULT_FILES[*]}' to $scheme location"
     Log "lftp -c open $OUTPUT_URL; mput ${RESULT_FILES[*]}"
-    lftp -c "open $OUTPUT_URL; mput ${RESULT_FILES[*]}"
-    StopIfError "Problem transferring files to $OUTPUT_URL"
+    lftp -c "open $OUTPUT_URL; mput ${RESULT_FILES[*]}" || Error "Problem transferring result files to $OUTPUT_URL"
     ;;
 
     (rsync)
     [[ "$BACKUP" = "RSYNC" ]] && return 0   # output/RSYNC/default/90_copy_result_files.sh took care of it
-    LogPrint "Copying files '${RESULT_FILES[@]}' to $scheme location"
+    LogPrint "Copying result files '${RESULT_FILES[@]}' to $scheme location"
     Log "rsync -a $v ${RESULT_FILES[@]} ${host}:${path}"
-    rsync -a $v "${RESULT_FILES[@]}" "${host}:${path}"
-    StopIfError "Problem transferring files to $OUTPUT_URL"
+    rsync -a $v "${RESULT_FILES[@]}" "${host}:${path}" || Error "Problem transferring result files to $OUTPUT_URL"
     ;;
 
-    (*) BugError "Support for $scheme is not implemented yet."
+    (*) Error "Invalid scheme '$scheme' in '$OUTPUT_URL'."
```

Comment:
Why should be unknown schemes considered invalid? In https://github.com/rear/rear/blob/3a6993968852791a63fac49bce6e81d52e258622/usr/share/rear/lib/global-functions.sh#L557 unknown schemes are passed as the `-t` parameter to `mount`, so if it supports the scheme/filesystem type, there should be no problem.

Reply:
Thinking about it more: the code above will work only for filesystems whose mount command line syntax is "NFS-like": host:path. Not many filesystem types will support that. Still, it may be better to leave supported filesystem handling to `mount_url()`, which errors out properly if the URL can not be mounted, and omit it from here (thanks to https://github.com/rear/rear/blob/3a6993968852791a63fac49bce6e81d52e258622/usr/share/rear/lib/global-functions.sh#L563 this `Error "Invalid scheme` situation should be essentially unreachable).

Reply:
@pcahyna 
I don't know why it errors out there (there is no comment that explains why).

I had only replaced the `BugError` (i.e. abort when there is a bug in our code)
by `Error` (i.e. exit when there is an error - usually caused by the user or his environment).

Even if that `Error "Invalid scheme ..."` cannot be reached
I would like to keep it there because it is the default case handling.

I would prefer to always have default case handling code
to make it clear to others who read the code what the default case handling is.
In general I don't like `case` statements without default case.

Furthermore even if a default case cannot happen with current code
it may happen with future code so having default case handling
makes the code future proof (i.e. fail safe for future changes elsewhere).

FYI
where we have currently such `Error "Invalid scheme ..."` code:
```
# find usr/share/rear -type f | xargs grep -B1 'Invalid scheme '

usr/share/rear/output/default/950_copy_result_files.sh
    (*)
        Error "Invalid scheme '$scheme' in '$OUTPUT_URL'."

usr/share/rear/output/PXE/default/820_copy_to_net.sh
        ;;
    (*) Error "Invalid scheme '$scheme' in '$OUTPUT_URL'."

usr/share/rear/verify/YUM/default/050_check_YUM_requirements.sh
        (*)
            Error "Invalid scheme '$scheme' in BACKUP_URL '$BACKUP_URL' valid schemes: nfs cifs usb tape file iso sshfs ftpfs"

usr/share/rear/prep/NETFS/default/050_check_NETFS_requirements.sh
        (*)
            Error "Invalid scheme '$scheme' in BACKUP_URL '$BACKUP_URL' valid schemes: nfs cifs usb tape file iso sshfs ftpfs"

usr/share/rear/prep/DUPLICITY/default/210_check_NETFS_URL_requirements.sh
        (*)
            Error "Invalid scheme '$scheme' in BACKUP_DUPLICITY_NETFS_URL '$BACKUP_DUPLICITY_NETFS_URL' valid schemes: nfs cifs usb tape file iso sshfs ftpfs"
```

Reply:
> Even if that `Error "Invalid scheme ..."` cannot be reached
> I would like to keep it there because it is the default case handling.
> 
> I would prefer to always have default case handling code
> to make it clear to others who read the code what the default case handling is.
> In general I don't like `case` statements without default case.

The problem with this approach is the inconsistency with the default case handling in `mount_url()`, which handles unknown schemes just fine, if there is support in `mount` for them with the right syntax.

I understand the need for default case, but I think it should be changed to something like 
```
    (*)
        # mount_url has handled it for us
```
and continue with the same code as in the nfs branch (actually, one can remove the nfs branch entirely and let the default branch handle it).

Reply:
Anyway, I am working on a larger cleanup, so I will try to include something similar (as a separate commit, so it can be easily omitted, in case you don't like it).

Reply:
One more thought

> Furthermore even if a default case cannot happen with current code
> it may happen with future code so having default case handling
> makes the code future proof (i.e. fail safe for future changes elsewhere).

I think that if we determine that the default case is truly intended to be unreachable, the `Error` should be changed back to `BugError` since it means that some invariant in the code was violated. If then there is a code change which makes it legitimately happen again, it is the responsibility of the person who introduces the change to update the default branch, because we can't foresee now what the future change will be and how the default branch should look like.

Reply:
@jsmeix I implemented this in the latest commit ( 1f8ef6353bfcd93fc2e1e12642de8bfbc2a1dca7 ) in my branch ( https://github.com/pcahyna/rear/tree/fix-backup-removal-in-exit-task ).

Reply:
It is commit 696ce19 in PR #2625.